### PR TITLE
Mathod for parraler execution

### PR DESCRIPTION
**Pull request by Bemhreth**

### CODE REVIEW

File: user_data/strategies/Strategy002.py [hunk 1]
```diff
@@ -168,10 +168,8 @@ def confirm_trade_entry(self, pair: str, order_type: str, amount: float, rate: f
         mode = "test"
         coin = pair.split("/")[0]
         brain = "Freq_" + self.__class__.__name__
-        if IS_BACKTEST:
```
Comment (tef-github) [inline user_data/strategies/Strategy002.py:171]: need IS_BACK_TEST logic here

File: user_data/strategies/util.py [hunk 2]
```diff
@@ -1,21 +1,25 @@
...
-BACKTEST_MONTH_INDEX = 9
-IS_BACKTEST = False
-WORKSPACE_PATH = "workspace2" if IS_BACKTEST else "workspace"
-EXECUTION_PATH = "/root/" + WORKSPACE_PATH + "/execution/"
-
-
-def launcher(mode, coin, brain):
-    threading.Thread(target=_perform_launcher, args=(mode, coin, brain)).start()
+def launcher(mode, coin, brain,date_time = 12344315):
+    if Config.IS_BACKTEST:
```
Comment (tef-github) [inline user_data/strategies/util.py]: this is wrong, backtest shouldn't point to _perform_launcher

create _perform_backtest()

File: user_data/strategies/Strategy002.py (hunk 1 above)
Comment (Bemhreth) [reply user_data/strategies/Strategy002.py:171]: I took care of that inside the utill

Comment (Bemhreth) [reply user_data/strategies/Strategy002.py:171]: As you suggested to make the code in strategy as clean as possible

**Review by tef-github — CHANGES_REQUESTED** [review]
changes requested

File: user_data/strategies/Strategy002.py (hunk 1 above)
Comment (tef-github) [reply user_data/strategies/Strategy002.py:171]: yes, if IS_BACKTEST -> backtest else launcher

**Comment by tef-github** [comment]
@Bemhreth changes missing in Strategy 3 and 4

**Comment by Bemhreth** [comment]
> @Bemhreth changes missing in Strategy 3 and 4

Fixed

File: user_data/strategies/Strategy002.py (hunk 1 above)
Comment (Bemhreth) [reply user_data/strategies/Strategy002.py:171]: @tef-github this is done in util.py as I mentioned so should I consider it as resolved?

File: user_data/strategies/util.py [hunk 3]
```diff
@@ -1,21 +1,25 @@
...
     date = str(date_time)
     date = date.replace(" ", "#")
-    subprocess.call("python3 "+ EXECUTION_PATH + "back_tester.py " + date + " " + coin + " " + brain + " 0.45 3", shell=True)
+    subprocess.call("python3 "+ Config.EXECUTION_PATH + "back_tester.py " + date + " " + coin + " " + brain + " 0.45 3", shell=True)
+
+def back_test(date_time, coin, brain):
+    if Config.IS_PARRALER_EXECUTION:
+        threading.Thread(target=_perform_back_test, args=(coin, brain, date_time)).start()
+    else:
+        date = str(date_time)
```
Comment (tef-github) [inline user_data/strategies/util.py]: you can call _perform_back_test here instead

File: user_data/strategies/util.py [hunk 4]
```diff
@@ -1,21 +1,25 @@
...
-BACKTEST_YEAR = 2020
-BACKTEST_MONTH_INDEX = 9
-IS_BACKTEST = False
-WORKSPACE_PATH = "workspace2" if IS_BACKTEST else "workspace"
-EXECUTION_PATH = "/root/" + WORKSPACE_PATH + "/execution/"
+def launch(mode, coin, brain):
+    if Config.IS_PARRALER_EXECUTION:
+        threading.Thread(target=_perform_launch, args=(mode, coin, brain)).start()
+    else:
+        subprocess.call("python3 "+Config.EXECUTION_PATH+"launcher.py " + mode + " " + coin + " " + brain, shell=True)
```
Comment (tef-github) [inline user_data/strategies/util.py]: you can call _perform_execute here

**Review by tef-github — CHANGES_REQUESTED** [review]
changes requested

**Review by tef-github — APPROVED** [review]

**Pull request merged**